device/cluster.cpp

```diff
@@ -171,11 +171,14 @@ struct tt_4_byte_aligned_buffer {
 namespace tt::umd {
 
 bool Cluster::address_in_tlb_space(
-    uint32_t address, uint32_t size_in_bytes, int32_t tlb_index, uint64_t tlb_size, std::uint32_t chip) {
-    return (
-        (tlb_config_map.at(chip).find(tlb_index) != tlb_config_map.at(chip).end()) &&
-        address >= tlb_config_map.at(chip).at(tlb_index) &&
-        (address + size_in_bytes <= tlb_config_map.at(chip).at(tlb_index) + tlb_size));
+    uint64_t address, uint32_t size_in_bytes, int32_t tlb_index, uint64_t tlb_size, std::uint32_t chip) {
+    const auto& tlb_map = tlb_config_map.at(chip);
+    const auto it = tlb_map.find(tlb_index);
```

Comment:
I rewrote this so that it is more explicit about the number of lookups that need to be performed: 2.